support/support.go

```diff
@@ -296,6 +279,121 @@ func WorkspaceInfo(ctx context.Context, client *codersdk.Client, log slog.Logger
 	return w
 }
 
+func AgentInfo(ctx context.Context, client *codersdk.Client, log slog.Logger, agentID uuid.UUID) Agent {
+	var (
+		a  Agent
+		eg errgroup.Group
+	)
+
+	if agentID == uuid.Nil {
+		log.Error(ctx, "no agent id specified")
+		return a
+	}
+
+	eg.Go(func() error {
+		agt, err := client.WorkspaceAgent(ctx, agentID)
+		if err != nil {
+			return xerrors.Errorf("fetch workspace agent: %w", err)
+		}
+		a.Agent = agt
+		return nil
+	})
+
+	eg.Go(func() error {
+		agentLogCh, closer, err := client.WorkspaceAgentLogsAfter(ctx, agentID, 0, false)
+		if err != nil {
+			return xerrors.Errorf("fetch agent startup logs: %w", err)
+		}
+		defer closer.Close()
+		var logs []codersdk.WorkspaceAgentLog
+		for logChunk := range agentLogCh {
+			logs = append(logs, logChunk...)
+		}
+		a.StartupLogs = logs
+		return nil
+	})
+
+	dialCtx, dialCancel := context.WithCancel(ctx)
+	defer dialCancel()
+	conn, err := client.DialWorkspaceAgent(dialCtx, agentID, &codersdk.DialWorkspaceAgentOptions{
+		Logger:         log.Named("dial-agent"),
+		BlockEndpoints: false,
+	})
+	if err != nil {
+		log.Error(ctx, "dial agent", slog.Error(err))
+	} else {
+		defer func() {
+			if err := conn.Close(); err != nil {
+				log.Error(ctx, "failed to close agent connection", slog.Error(err))
+			}
+			<-conn.Closed()
+		}()
+		if !conn.AwaitReachable(ctx) {
+			log.Error(ctx, "timed out waiting for agent")
+		} else {
+			eg.Go(func() error {
+				_, _, pingRes, err := conn.Ping(ctx)
+				if err != nil {
+					return xerrors.Errorf("ping agent: %w", err)
+				}
+				a.PingResult = pingRes
+				return nil
+			})
+
+			eg.Go(func() error {
+				pds := conn.GetPeerDiagnostics()
+				a.PeerDiagnostics = &pds
+				return nil
+			})
+
+			eg.Go(func() error {
+				msBytes, err := conn.DebugMagicsock(ctx)
+				if err != nil {
+					return xerrors.Errorf("get agent magicsock page: %w", err)
+				}
+				a.MagicsockHTML = msBytes
+				return nil
+			})
+
+			eg.Go(func() error {
+				manifestRes, err := conn.DebugManifest(ctx)
+				if err != nil {
+					return xerrors.Errorf("fetch manifest: %w", err)
+				}
+				if err := json.NewDecoder(bytes.NewReader(manifestRes)).Decode(&a.Manifest); err != nil {
```

Comment:
Wondering, is there a need to decode this, actually? Or should we just use the bytes? Do we actually access any of the data here, or do we reformat it?

Reply:
We could probably just write the bytes. We can still validate in the tests that it is, in fact, a valid manifest.

Reply:
I forgot -- we do actually reformat for readability 😅 So keeping as-is.